src/hera/workflows/script.py

```diff
@@ -662,16 +655,9 @@ def script_wrapper(
 
         s = Script(name=name, source=source, **script_kwargs)
 
-        @overload
-        def task_wrapper(*args: FuncIns.args, **kwargs: FuncIns.kwargs) -> FuncR:  # type: ignore
-            ...
-
-        @overload
-        def task_wrapper(*args: ScriptIns.args, **kwargs: ScriptIns.kwargs) -> Union[Step, Task]: ...
-
         @wraps(func)
-        def task_wrapper(*args, **kwargs):
-            """Invokes a `Script` object's `__call__` method using the given `task_params`."""
+        def task_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
```

Comment:
Why did we remove wraps decorator? This sets a bunch of important attributes correctly.

Reply:
It's still there

![image](https://github.com/argoproj-labs/hera/assets/17798778/53f5178e-aa7f-4869-be3b-e440cd950b85)